syntax = "proto2";

package classification.protos;

message InputReader {
  // Path to StringIntLabelMap pbtxt file specifying the mapping from string
  // labels to integer ids.
  optional string label_map_path = 1 [default=""];

  // Whether data should be processed in the order they are read in, or
  // shuffled randomly.
  optional bool shuffle = 2 [default=true];

  //Number of examples to keep in queue at min (to allow for good shuffling)
  //min_after_dequeue is calculated using this. (only if shuffling or batched eval)
  optional float fraction_of_examples_in_queue = 3 [default=0.08];

  //how many batches past the min_queue_size to keep in queue (only if shuffling or batched eval)
  optional int32 num_batches_past_min_queue_size = 4 [default=3];

  // The number of times a data source is read. If set to zero, the data source
  // will be reused indefinitely.
  optional uint32 num_epochs = 5 [default=0];

  // Number of threads to batch data.
  optional uint32 num_threads = 6 [default=4];

  required int32 batch_size = 7;

  oneof input_reader {
    TFRecordInputReader tf_record_input_reader = 8;
    ExternalInputReader external_input_reader = 9;
  }

  repeated string multi_task_label_name = 10;

  required uint32 image_height = 11;
  required uint32 image_width = 12;

  //only used by an eval input config
  optional bool eval_batch_mode = 13 [default=false];

  required int64 num_examples = 14;
}

// An input reader that reads TF Example protos from local TFRecord files.
message TFRecordInputReader {
  // Path(s) to `TFRecordFile`s.
  required string input_path = 1;
}

// An externally defined input reader. Users may define an extension to this
// proto to interface their own input readers.
message ExternalInputReader {
  extensions 1 to 999;
}
